### PR TITLE
fix double quote problem when we parse indexTarget

### PR DESCRIPTION
**Pull request by Yuqi-Du**
This is a redo for [#1723](https://github.com/stargate/data-api/pull/1772)

What this PR does:
When columnName is doubleQuoted in index target, previous regex can not parse it correctly, which will end up create an unknown ApiIndex.
This will not only cause confusion in listIndexCommand, but also for any command using that index, will complain index not found, because it is parsing as  an unknown unsupported index.

Which issue(s) this PR fixes:
Fixes [#1770]

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java [hunk 1]
```diff
@@ -149,4 +157,210 @@ public void listIndexesWithDefinition() {
...
+              "CREATE CUSTOM INDEX IF NOT EXISTS idx_full_frozen_map ON \"%s\".\"%s\" (FULL(\"frozenMapColumn\")) USING 'StorageAttachedIndex'",
+              keyspaceName, PRE_EXISTED_CQL_TABLE);
+      assertThat(executeCqlStatement(SimpleStatement.newInstance(fullIndex))).isTrue();
+    }
+
+    @Test
+    @Order(2)
+    public void listIndexesWithDefinition() {
+      // TODO, currently ApiIndexType treats index on map/set/list as unsupported, so the index on
+      // these columns have UNKNOWN column in the definition
```
Comment (Hazel-Datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java:250]: Only `full` is not supported and only `UNKNOWN` in `full` index column?

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java [hunk 2]
```diff
@@ -149,4 +157,210 @@ public void listIndexesWithDefinition() {
                   jsonEquals(createVectorIndex)));
     }
   }
+
+  // ==================================================================================================================
+  // This subClass is to test some index corner cases.
+  // 1. Currently, Data API does not support create index on set/list/map column, so we need to
```
Comment (Hazel-Datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java:163]: `Data API does not support create index on set/list/map column`
does not -> does?

**Review by Hazel-Datastax — COMMENTED** [review]
LGTM overall

**Review by Hazel-Datastax — APPROVED** [review]

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java (hunk 1 above)
Comment (Yuqi-Du) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java]: updated, yes, full index on frozen map is not supported, will be mapped into UNKNOWN.
Detail in [here](https://github.com/stargate/data-api/blob/main/src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/indexes/UnsupportedIndexDefinitionDesc.java)

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java (hunk 2 above)
Comment (Yuqi-Du) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java]: fixed

File: src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CQLSAIIndex.java [hunk 3]
```diff
@@ -128,27 +127,38 @@ static IndexTarget indexTarget(IndexMetadata indexMetadata)
...
-        ? new IndexTarget(CqlIdentifier.fromInternal(columnName), null)
-        : new IndexTarget(
-            CqlIdentifier.fromInternal(columnName), ApiIndexFunction.fromCql(functionName));
+    // 1. In the case of a quoted column name the name in the target string
+    // will be enclosed in quotes, which we need to unwrap.
+    // 2. It may also include quote characters internally, escaped like so:
+    // abc"def -> abc""def, then we need to un-escape as abc"def -> abc"def
+    // to get the actual column name
+    if (columnName.startsWith(QUOTE)) {
+      columnName = StringUtils.substring(StringUtils.substring(columnName, 1), 0, -1);
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CQLSAIIndex.java:158]: Could avoid extra call (and dep to `StringUtils`) by using single:

   columnName = columnName.substring(1, columnName.length() - 1);

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java [hunk 4]
```diff
@@ -149,4 +157,210 @@ public void listIndexesWithDefinition() {
...
+  // This subClass is to test some index corner cases.
+  // 1. Currently, Data API does not support create index on frozen collection, also does not
+  // support create FULL index on frozen map.
+  // 2. Data API resolves index target from IndexMetaData, detail in {@link CQLSAIINDEX}, test
+  // columns with doubleQuote
+  // ==================================================================================================================
+  @Nested
+  @Order(2)
+  public class CreatedIndexOnPreExistedCqlTable {
+    private static final String PRE_EXISTED_CQL_TABLE = "perExistedCqlTable";
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/ListIndexesIntegrationTest.java:171]: nit: perExisted -> preExisted

**Review by tatu-at-datastax — APPROVED** [review]
LGTM -- couple of minor optional suggestions.

File: src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CQLSAIIndex.java (hunk 3 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CQLSAIIndex.java:158]: thanks, changed.
Wonder why cassandra wrote the other way..

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CQLSAIIndex.java]: Yeah don't know -- doing 2 calls results in 2 separate String allocations which is wasteful (obv. not a huge deal, just unnecessary).
Could even offer a patch for Cassandra (esp. if occurs in many places).

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:
